sql-map-jquery-test/src/main/java/com/hazelcast/jet/tests/jquery/tests/AbstractJsonMapTest.java

```diff
@@ -0,0 +1,174 @@
+    protected final String sqlQuery;
+    protected final Boolean requiredSort;
+    protected final String expectedJqueryResultString;
+    protected HazelcastInstance client;
+    protected int queryTimeout = propertyInt("queryTimeout", DEFAULT_QUERY_TIMEOUT_MILLIS);
+    protected String jsonInputString;
+    private long begin;
+    private long currentQueryCount;
+    private long lastQueryCount;
+    private long lastProgressPrintCount;
+
+    public AbstractJsonMapTest(String mapName, String inputJsonFile, String sqlQuery, Boolean requiredSort,
+                               String expectedJsonPath, Boolean resultIsArray) throws IOException, URISyntaxException {
+        this.mapName = mapName;
+        this.jsonInputString = readJsonFromFile(inputJsonFile);
+        this.sqlQuery = sqlQuery;
+        this.requiredSort = requiredSort;
+        this.expectedJqueryResultString = retrieveExpectedJsonStructure(jsonInputString, expectedJsonPath,
+                resultIsArray);
+    }
+
+    protected abstract String retrieveExpectedJsonStructure(String expectedJsonInputString, String expectedJsonPath,
+                                                            Boolean resultIsArray);
+
+    protected void runTest() {
+        begin = System.currentTimeMillis();
+
+        SqlService sql = client.getSql();
+
+        String sqlString = "CREATE OR REPLACE MAPPING " + mapName +
+                " TYPE " + IMapSqlConnector.TYPE_NAME + "\n" +
+                "OPTIONS (\n" + '\''
+                + OPTION_KEY_FORMAT + "'='" + "bigint" + "'\n" + ", '"
+                + OPTION_VALUE_FORMAT + "'='" + "json" + "'\n" + ")";
+
+        try (SqlResult result = client.getSql().execute(sqlString)) {
+            assertEquals(result.updateCount(), 0L);
+        }
+
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+
+            //Execute query
+            SqlResult sqlResult = sql.execute(sqlQuery);
+
+            //Check that query returned results
+            assertTrue("The following query was not successful: " + sqlQuery, isQuerySuccessful(sqlResult,
+                    expectedJqueryResultString));
+            currentQueryCount++;
+
+            //Print progress
+            printProgress();
+
+            //Timeout between queries to not stress out the cluster
+            Util.sleepMillis(queryTimeout);
+        }
+        logger.info(String.format("Test completed successfully. Executed %d queries in %s.", currentQueryCount,
+                getTimeElapsed()));
+    }
+
+    private void printProgress() {
+        long nextPrintCount = lastProgressPrintCount + PROGRESS_PRINT_QUERIES_INTERVAL;
+        boolean toPrint = currentQueryCount >= nextPrintCount;
+        if (toPrint) {
+            logger.info(String.format("Time elapsed: %s. Executed %d queries", getTimeElapsed(), currentQueryCount));
+            lastProgressPrintCount = currentQueryCount;
+        }
+        assertNotStuck();
```

Comment:
Just a note - we've already discuss it on slack - we will need to refactor this in the whole SQL test suite.

Reply:
Yes, this will be a new refactor, we can have some sort of progress Monitor in separate thread. This mechanics behind that should be framework level separate thread util for use in all modules